numpy/core/src/multiarray/number.c

```diff
@@ -381,53 +283,37 @@ array_inplace_right_shift(PyArrayObject *m1, PyObject *m2);
 static PyObject *
 array_add(PyArrayObject *m1, PyObject *m2)
 {
-    PyObject * res;
-    GIVE_UP_IF_HAS_RIGHT_BINOP(m1, m2, "__add__", "__radd__", 0, nb_add);
-    if (try_binary_elide(m1, m2, &array_inplace_add, &res, 1)) {
```

Comment:
Definitely unintended. Looking at the commits, this happened in the initial rebase of Nathaniel's work that made the binops call the ufuncs (https://github.com/numpy/numpy/pull/8247/commits/73a4fc3f4a8da61a0cd81147d7d2bc1f436983f5#diff-ed156d214ca580fa436c7859cd48d21c). That predates the elision, and I must have missed that it was something new. Sorry.